src/test/java/htsjdk/samtools/cram/compression/rans/RansTest.java

```diff
@@ -49,107 +57,145 @@ public String toString() {
         };
     }
 
-    @Test(dataProvider="ransData")
-    public void testRANS(final TestCaseWrapper tc) {
-        roundTripForEachOrder(tc.testArray);
+    public Object[][] getRansTestDataTinySmallLarge() {
+        return new Object[][]{
+                { new TestCaseWrapper(randomBytesFromGeometricDistribution(100, 0.1)), 1, 100 }, // Tiny
+                { new TestCaseWrapper(randomBytesFromGeometricDistribution(1000, 0.01)), 4, 1000 }, // Small
+                { new TestCaseWrapper(randomBytesFromGeometricDistribution(100 * 1000 + 3, 0.01)), 100 * 1000 + 3 - 4, 100 * 1000 + 3 } // Large
```

Comment:
Would be great to document what these magic numbers mean (3, 4 ?) , if anything. It seems like there is some relationship between the args to `randomBytesFromGeometricDistribution` (since they appear there as well). Not sure if you know, but ideally we should figure it out and give them names.